src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -849,7 +882,12 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
             for (UserRemoteConfig uc : getUserRemoteConfigs()) {
                 String ucCredentialsId = uc.getCredentialsId();
                 String url = getParameterString(uc.getUrl(), environment);
-                chooser = new GitToolChooser(url, project, ucCredentialsId, gitTool, n, listener,unsupportedCommand.determineSupportForJGit());
+                /* GitPublisher doesn't support JGit, it should not be suggested */
+                if (!cmd.determineSupportForJGit()) {
+                    chooser = new GitToolChooser(url, project, ucCredentialsId, gitTool, n, listener,false);
+                } else {
+                    chooser = new GitToolChooser(url, project, ucCredentialsId, gitTool, n, listener, unsupportedCommand.determineSupportForJGit());
+                }
```

Comment:
I needed to study this carefully before I understood what it is expressing.

As far as I can tell, it says that if the post build action does not support JGit or any of the extensions do not support JGit, then the chooser will not choose JGit.

Could the variable `cmd` be named more clearly as `postBuildUnsupportedCommand` or something like that?  I was initially confused by the reference to `cmd.determineSupportForJGit` and to `unsupportedCommand.determineSupportForJGit`./

Optionally, I think it could be expressed as a single statement:

```suggestion
                chooser = new GitToolChooser(url, project, ucCredentialsId, gitTool, n, listener,
                                                                   cmd.determineSupportForJGit() && unsupportedCommand.determineSupportForJGit());
```